mmv1/third_party/tgc/imagewrapper.go

```diff
@@ -0,0 +1,23 @@
+package compute
+
+import (
+	tpg_services_compute "github.com/hashicorp/terraform-provider-google/google/services/compute"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+// ResolveImageWrapper is a wrapper around compute.ResolveImage and recover
+// from panic if client is il.
+func ResolveImageWrapper(c *transport_tpg.Config, project, name, userAgent string) (ret string, err error) {
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
ah - when I said "recover from the error" I meant that since you had already made the ResolveImage function return errors instead of panicking, you could check if the returned error indicates that the problem was that there's no client. Using `recover()` is not recommended. go/go-style/best-practices#checks-and-panics Sorry for the confusion!